### PR TITLE
feat(IdentityCenter): add identity center groups data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add identity center groups data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add identity center groups data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
go test ./huaweicloud/services/acceptance/identitycenter -v -run TestAccDatasourceIdentityCenterGroups_basic -timeout 360m
=== RUN   TestAccDatasourceIdentityCenterGroups_basic
=== PAUSE TestAccDatasourceIdentityCenterGroups_basic
=== CONT  TestAccDatasourceIdentityCenterGroups_basic
--- PASS: TestAccDatasourceIdentityCenterGroups_basic (10.83s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    15.592s
```

### CODE REVIEW

File: docs/incubating/identitycenter_groups.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `groups` - Indicates the list of IdentityCenter group.
+  The [Group](#IdentityCenterGroups_Group) structure is documented below.
+
+<a name="IdentityCenterGroups_Group"></a>
+The `Group` block supports:
```
Comment (Lance52259) [inline docs/incubating/identitycenter_groups.md]:
```suggestion
The `groups` block supports:
```

Comment (houpeng80) [reply docs/incubating/identitycenter_groups.md]: done

**Pull request merged**